tests/wpunit/ConnectionRegistrationTest.php

```diff
@@ -677,4 +679,45 @@ protected function assertValidTypes( $actual ) : void {
 		$this->assertEquals( 'TestObject', $actual['data']['testConnection']['nodes'][0]['__typename'] );
 	}
 
+	public function testRegisteringConnectionsWithCustomQueryClasses() {
```

Comment:
We should also have tests for some common edge cases: bad class name, incompatible class, and using the wrong `WP_{$type}_Query` are the ones that jump to mind.